lib/constructs/provisioning-sfn-workflow.ts

```diff
@@ -45,18 +46,36 @@ export interface IProvisioningWorkflow {
  * composed to form a workflow for provisioning jobs from SNOW to the CSPs.
  *
  */
-export class ProvisioningWorkflow implements IProvisioningWorkflow {
+export class ProvisioningWorkflow extends Construct implements IProvisioningWorkflow {
   readonly mappedTasks: TasksMap = {};
   readonly workflow: sfn.IChainable;
   readonly logGroup: logs.ILogGroup;
+  readonly provisioningJobsQueue: sqs.IQueue;
+  readonly resultFn: lambdaNodeJs.NodejsFunction;
 
-  constructor(scope: Construct, props: ProvisioningWorkflowProps) {
+  constructor(scope: Construct, id: string, props: ProvisioningWorkflowProps) {
+    super(scope, id);
     const { environmentName } = props;
+
+    this.provisioningJobsQueue = new sqs.Queue(scope, "ProvisioningJobsQueue", {
+      queueName: "ProvisioningJobsQueue",
```

Comment:
Do we _really_ want to name this? Won't this result in us only being able to build this once?